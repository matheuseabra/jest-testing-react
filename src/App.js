import React, { Component } from 'react';
import './App.css';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = { isLoaded: false, data: [] }
  }

  loadFromAPI() {
    fetch()
  }

  componentDidMount() {
    //this.loadFromAPI();
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h3>
            React Jest Testing 101
          </h3>  
        </header>
        <main>
          <div className="grid">
            <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
            </p>
          </div>
        </main>
      </div>
    );
  }
}

export default App;
