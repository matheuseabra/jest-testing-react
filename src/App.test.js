import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('title to contain React Jest Testing 101', () => {
  const element = renderer.create(<h3>
    React Jest Testing 101
  </h3>);

  expect(element).toMatchSnapshot();
});


