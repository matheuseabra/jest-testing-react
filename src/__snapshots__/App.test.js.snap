// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title to contain React Jest Testing 101 1`] = `
<h3>
  React Jest Testing 101
</h3>
`;
